src/libraries/Common/src/Interop/Unix/System.Native/Interop.Socket.cs

```diff
@@ -3,15 +3,18 @@
 
 using System;
 using System.Net;
-using System.Net.Internals;
 using System.Net.Sockets;
 using System.Runtime.InteropServices;
 
 internal static partial class Interop
 {
     internal static partial class Sys
     {
+#if SYSTEM_NET_SOCKETS_DLL
         [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_Socket")]
         internal static unsafe partial Error Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType, IntPtr* socket);
+#endif
+        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_Socket")]
+        internal static unsafe partial Error Socket(int addressFamily, int socketType, int protocolType, IntPtr* socket);
```

Comment:
Do we need both?  Should we just always use the one that uses ints and cast to int from AddressFamily/SocketType/ProtocolType when we have them?

Reply:
updated + cast in SocketPal